### ISSUE TITLE
Planning story: Comfortstat prototype

### ISSUE THREAD

**Issue by lmas** [lmas/d0020e_code#5]
My dudes, please leave a paper trail.

Example:
- [x] Document the use case, see slide L06.1 pg. 14 (and to the end) for examples
- [x] Draw a diagram to get an overview of what's required to implement?
- [x] Split up the use case into stories, document each (in report and as github issues?). See slide L2.1 pg. 11 for an example.
- [x] Split each story into tasks, create issues for each and estimate work time?
- [x] Prioritise the stories/tasks and put them on the board
- [x] Pick one story and start working on the highest prio. task

_Use Case 1_

**Name**: Sending setpoint(desired temperature) to Zigbee
**Participating actors**: Zigbee
**Entry condition**: Every time user changes min/max or every hour as a standard
**Exit condition**: An ACK has been received from Zigbee
**Event flow**: 

Each loop (interaction)
- Contact the orchestrator to receive knowledge of where to find Zigbee
- Orchestrator contacts the sregistrar for information about the state of Zigbee and URL
- If Zigbee is available an URL is send back to the comfortstat
- Calculations are performed with the max/min price and temperature intervals together with the hourly electric price to get the setpoint.
- Use the URL to send the calculated setPoint to Zigbee

_Use Case 2_

**Name**:  Handles new max/min temperature  values
**Participating actors**: User host
**Entry condition**: Receives a new max/min interval from user
**Exit condition**: Has changed the old min/max against the new ones and implemented them.  (saved them where???)
**Event flow**: 

- The user updates the preferable temperature intervals with two new values (max/min)

_Use Case 3_

**Name**:  Handles new max/min price values
**Participating actors**: User host
**Entry condition**: Receives a new max/min interval from user
**Exit condition**: Has changed the old min/max against the new ones and implemented them.  (saved them where???)
**Event flow**: 

- The user updates the preferable price range with two new values (max/min)

_Use Case Diagram_

![image](https://github.com/user-attachments/assets/92dbef8d-cbef-433b-92e0-6bb26683e6a5)

**Comment by lmas** [comment]
Before I forget, as I already did yesterday, have you considered Jan's latest mail about associating a cost with a setpoint? It sounded most relevant to your system.

**Comment by gabaxh** [comment]
Yes, at least I have read it and we will take it into consideration when building this system.

**Comment by SimonPergel** [comment]
```mermaid
sequenceDiagram
loop every x seconds
    comforstat->>+sregistrar: register
    sregistrar->>+comforstat: OK
end
 loop every time user change or ever hour
    comforstat->>+orchastrator: Where Zigbee?
    orchastrator->>+sregistrar: query
    sregistrar->>+orchastrator: URL
    orchastrator->>+comforstat: URL
    
   
    comforstat->>+Zigbee: setpoint
    end
```
@vanDeventer, hi Jan! This is how we have interpreded what we discussed last time, do you have any objections?

**Comment by SimonPergel** [comment]
_Use Case 1_

**Name**: Sending setpoint(desired temperature) to Zigbee
**Participating actors**: Zigbee
**Entry condition**: Every time user changes min/max or every hour as a standard
**Exit condition**: An ACK has been received from Zigbee
**Event flow**: 

Each loop (interaction)
- Contact the orchestrator to receive knowledge of where to find Zigbee
- Orchestrator contacts the sregistrar for information about the state of Zigbee and URL
- If Zigbee is available an URL is send back to the comfortstat
- Calculations are performed with the max/min price and temperature intervals together with the hourly electric price to get the setpoint.
- Use the URL to send the calculated setPoint to Zigbee
